hastory/src/Hastory/Storage/HastoryPath.hs

```diff
@@ -0,0 +1,22 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Hastory.Storage.HastoryPath where
+
+import Prelude
+import qualified Data.Text as T
+import Path (Path, Abs, Dir, toFilePath, parseAbsDir)
+import Database.Persist (PersistValue(..))
+import Database.Persist.Sql
+
+newtype HastoryPath = HastoryPath { unHastoryPath :: Path Abs Dir }
```

Comment:
This is fine, and you can leave it as-is, but for the record: orphan instances don't _really_ matter as much _in an application_.

Reply:
It was getting pretty hairy to make the rest of the modules deal with the newtype. Instead, I created the `Hastory.Types.Path` module to hold the orphaned instances.